Lib/getpass.py

```diff
@@ -68,12 +71,37 @@ def unix_getpass(prompt='Password: ', stream=None):
                 old = termios.tcgetattr(fd)     # a copy to save
                 new = old[:]
                 new[3] &= ~termios.ECHO  # 3 == 'lflags'
+                if echochar:
+                    new[3] &= ~termios.ICANON
                 tcsetattr_flags = termios.TCSAFLUSH
                 if hasattr(termios, 'TCSASOFT'):
                     tcsetattr_flags |= termios.TCSASOFT
                 try:
                     termios.tcsetattr(fd, tcsetattr_flags, new)
-                    passwd = _raw_input(prompt, stream, input=input)
+                    if not echochar:
+                        passwd = _raw_input(prompt, stream, input=input)
+                        stream.write('\n')
+                        return passwd
+
+                    passwd = ""
+                    stream.write(prompt)
+                    stream.flush()
+                    while True:
+                        char = input.read(1)
+                        if char == '\n' or char == '\r':
+                            break
+                        if char == '\x03':
+                            raise KeyboardInterrupt
+                        if char == '\x7f' or char == '\b':
+                            if echochar and passwd:
+                                stream.write("\b \b" * len(echochar))
```

Comment:
IIUC, this tells termios to remove the last char right?

Reply:
right